widget/text.go

```diff
@@ -90,7 +103,7 @@ func (t *textProvider) updateRowBounds() {
 	textWrap := t.presenter.textWrap()
 	textStyle := t.presenter.textStyle()
 	textSize := theme.TextSize()
-	maxWidth := t.Size().Width - 2*theme.Padding()
+	maxWidth := t.size.Width - 2*theme.Padding()
```

Comment:
The func doc should explain what lock callers must hold

Reply:
In this instance I thought it was clear - the method was an exported method so is safe (locked). As I am obtaining the lock I can access the field but not call the method that would deadlock.

Reply:
Yup, I mean updateRowBounds' doc should explain that callers hold lock

Reply:
Thanks. This is an ongoing discussion about documentation that I don't think has been resolved yet.